wallet-core/src/lib.rs

```diff
@@ -62,13 +62,18 @@ pub fn map_owned(
 /// [`PhoenixViewKey`].
 pub fn phoenix_balance(
     phoenix_vk: &PhoenixViewKey,
-    unspent_notes: impl AsRef<[Note]>,
+    notes: impl AsRef<[EnrichedNote]>,
```

Comment:
Why don't you just implement `AsRef<Note> for Note`?

Reply:
After merging #2248 we can even use the `NoteLeaf` from execution-core and with that we probably can leverage the `From` trait for a conversion from a `NoteLeaf` to a `Note`